### DISCUSSION TITLE
Monitor conditions (avoidance synchronisation)

### DISCUSSION THREAD

**Discussion by mwallnoefer** [jolie/jolie#504] [Ideas (Features, Fixes, etc.)]
I wonder why Jolie does not provide any support for monitor condition variables like `wait()` and `notify()` respectively `signal()` in `synchronization` blocks. Has this support been excluded by design? Would they interfere with internal design choices?

**Comment by fmontesi** [comment]
Mostly to keep the language simple, as many synchronisation patterns can be implemented by embedding a (Java or Jolie) service and then interacting with it via communication.

Is there a particular pattern/use case/problem that you're interested in?

**Reply by mwallnoefer** [reply]
As I have studied the Jolie chat mqtt example, I have wondered if I could make the http fetch messages operation wait (maybe some sort of timed wait) until new messages from the Mqtt side come in. This would permit to implement some sort of long-polling strategy for the web browser.
